Sprint-2/stretch/count-words.js

```diff
@@ -26,3 +26,35 @@
 
 3. Order the results to find out which word is the most common in the input
 */
+
+function countWords (str){
+  // clean up the string
+  str = str.toLowerCase().replace(/[.,!?]/g, ""); // Remove all of the punctuation + convert to lowercase
+
+  const wordCounts = {};
+  const words = str.split(" "); // split it into words
+
+  str = str.toLowerCase().replace(/[.,!?]/g, ""); // Remove all of the punctuation + convert to lowercase
+
+  for (const word of words) {
+    if (word) {
+      // skip empty strings
+      wordCounts[word] = (wordCounts[word] || 0) + 1;
+    }
+  }
+  const orderedCounts = Object.fromEntries(
+    Object.entries(wordCounts).sort(([, a], [, b]) => b - a)
+    // Object.entries(wordCounts) converts obj > array of ky-value pairs
+    // .sort(([, a], [, b]) => b - a) sorts by value
+    // Object.fromEntries converts the sorted array back into obj 
+  );
+
+  return orderedCounts;
+}
+
```

Comment:
What is the return value you expect from the following function call?
Does your function return the value you expected?
```
countWords("A.A A A_A A$A")
```